src/client/java/teammates/client/scripts/DataMigrationForSanitizedInstructorName.java

```diff
@@ -0,0 +1,69 @@
+package teammates.client.scripts;
+
+import java.io.IOException;
+
+import com.googlecode.objectify.Key;
+import com.googlecode.objectify.cmd.Query;
+
+import teammates.common.util.SanitizationHelper;
+import teammates.storage.entity.Instructor;
+
+/**
+ * Script to desanitize name of {@link Instructor} if it is sanitized.
+ */
+public class DataMigrationForSanitizedInstructorName
+        extends DataMigrationEntitiesBaseScript<Instructor> {
+
+    public DataMigrationForSanitizedInstructorName() {
+        numberOfScannedKey.set(0L);
+        numberOfAffectedEntities.set(0L);
+        numberOfUpdatedEntities.set(0L);
+    }
+
+    public static void main(String[] args) throws IOException {
+        DataMigrationForSanitizedInstructorName migrator =
+                new DataMigrationForSanitizedInstructorName();
+        migrator.doOperationRemotely();
+    }
+
+    @Override
+    protected Query<Instructor> getFilterQuery() {
+        return ofy().load().type(Instructor.class);
+    }
+
+    @Override
+    protected boolean isPreview() {
+        return true;
+    }
+
+    @Override
+    protected String getLastPositionOfCursor() {
+        return "";
+    }
+
+    @Override
+    protected int getCursorInformationPrintCycle() {
+        return 100;
+    }
+
+    @Override
+    protected boolean isMigrationNeeded(Key<Instructor> key) throws Exception {
+        Instructor instructor = ofy().load().key(key).now();
+
+        if (!SanitizationHelper.sanitizeForHtml(instructor.getName()).equals(instructor.getName())) {
+            System.err.println(String.format("Instructor %s has unsanitized name %s, this should not happen",
```

Comment:
Actually, name such as `John Doe` is the same before and after sanitisation. It is not an error.

Reply:
There's a ! in front to check for not equals. The goal of this is to check if the name changed during sanitization.

Reply:
But `isMigrationNeeded` is not run after `migrateEntity`.

Reply:
Sorry, I mispoke. Since isMigrationNeeded is run before, and all current code sanitizes input before saving it into the database, all current records should be sanitized. Thus, this code checks if there are any records that weren't sanitized (by sanitizing it and checking for changes) and outputs an error message if it finds that it wasn't sanitized.

Looking in the other classes, I'm kind of confused now on their implementation. DataMigrationForSanitizedDataInInstructorAttributes.isMigrationNeeded checks for: `SanitizationHelper.isSanitizedHtml(instructor.getRole())`, and prints that the "Instructor has unsanitized role" if it is true. But if that returns true, then it means it that was sanitized...

Reply:
Ohh. I think you are mislead by other scripts :P.

`DataMigrationForSanitizedDataInInstructorAttributes` is written long time ago and has not been run (don't know why). The original script do de-sanitization to `role`. However `role` should be a set of fixed values and there is no need to sanitise/de-sanitise it. With such confusion, when reviewing the script,  we remove the de-sanitization for `role` and add a check for it (just for safe in case we are missing something).

Reply:
Ah, ok. Should I remove this check then? Or just leave it in?

Reply:
Just remove then.